cpp/include/cudf/detail/aggregation.hpp

```diff
@@ -50,6 +52,117 @@ struct quantile_aggregation : aggregation {
+template <> struct corresponding_operator<aggregation::MAX> { using type = DeviceMax; };
+template <> struct corresponding_operator<aggregation::SUM> { using type = DeviceSum; };
+template <> struct corresponding_operator<aggregation::COUNT> { using type = DeviceSum; };
+template <aggregation::Kind k>
+using corresponding_operator_t = typename corresponding_operator<k>::type;
+
+/**---------------------------------------------------------------------------*
+ * @brief Determines accumulator type based on input type and aggregation.
+ *
+ * @tparam SourceType The type on which the aggregation is computed
+ * @tparam k The aggregation performed
+ *---------------------------------------------------------------------------**/
+template <typename SourceType, aggregation::Kind k, typename Enable = void>
+struct target_type_impl { using type = void; };
+
+// Computing MIN of SourceType, use SourceType accumulator
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MIN> { using type = SourceType; };
+
+// Computing MAX of SourceType, use SourceType accumulator
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MAX> { using type = SourceType; };
+
+// Always use size_type accumulator for COUNT
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::COUNT> { using type = cudf::size_type; };
+
+// Always use `double` for MEAN
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MEAN> { using type = double; };
+
+// Summing integers of any type, always use int64_t accumulator
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::SUM,
+                   std::enable_if_t<std::is_integral<SourceType>::value>> {
+  using type = int64_t;
+};
+
+// Summing float/doubles, use same type accumulator
+template <typename SourceType>
+struct target_type_impl<
+    SourceType, aggregation::SUM,
+    std::enable_if_t<std::is_floating_point<SourceType>::value>> {
+  using type = SourceType;
+};
+
+// Always use `double` for quantile 
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::QUANTILE> { using type = double; };
+
+// MEDIAN is a special case of a QUANTILE  
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MEDIAN> {
+   using type = target_type_impl<SourceType, aggregation::QUANTILE>; 
+};
+
+/**
+ * @brief Helper alias to get the accumulator type for performing aggregation 
+ * `k` on elements of type `SourceType`
+ *
+ * @tparam SourceType The type on which the aggregation is computed
+ * @tparam k The aggregation performed
+ */
+template <typename SourceType, aggregation::Kind k>
+using target_type_t = typename target_type_impl<SourceType, k>::type;
+
+/**
+ * @brief Dispatches  k as a non-type template parameter to a callable,  f.
+ *
+ * @tparam F Type of callable
+ * aram k The `aggregation::Kind` value to dispatch
```

Comment:
Looks like a search and delete of @p caused this...
```suggestion
 * @param k The `aggregation::Kind` value to dispatch
```